### PR TITLE
Feature/gitlab provider

### PR DESCRIPTION
**Pull request by Xyand**
Added GitLab provider. Tried to stick to the initial implementation (additional refactor wouldn't hurt)
Polling and app webhook not covered in this PR.
Tested on several group level (not user) GitLab repositories (using pr_agent/cli.py)

Some tests are failing however they're failing on main as well.

### CODE REVIEW

**Comment by uripeled2** [comment]
@CodiumAI-Agent

File: pr_agent/settings/.secrets_template.toml [hunk 1]
```diff
@@ -25,3 +25,10 @@ private_key = """\
 """
 app_id = 123456  # The GitHub App ID, replace with your own.
 webhook_secret = "<WEBHOOK SECRET>"  # Optional, may be commented out.
+
+[gitlab]
+# The type of deployment to create. Valid values are 'app' or 'user'.
```
Comment (Enase) [inline pr_agent/settings/.secrets_template.toml:30]: consider to update comment
```suggestion
# Gitlab personal access token
```

**Comment by okotek** [comment]
Thanks for your contribution! 
We'll need to make some adoptions and expect to merge early next week. There are some elements that needs to be refactored in order to decouple the git provider from the main logic, for example how to handle "installation id" which is relevant only for Github App and received as part of the webhook, as well as tests.

**Comment by Xyand** [comment]
Hi @okotek. I added a gitlab polling and took care to some extent of separating installation ID in this branch (the name is misleading): https://github.com/Xyand/pr-agent/tree/feature/gitlab_webhook

LMK if you want me to add it to this PR or add it separately.

**Comment by okotek** [comment]
Thanks @Xyand. 
Add it all in the same PR, it's all related to gitlab support. We have a small backlog but we'll merge this as soon as possible

**Comment by okotek** [comment]
The main issue is injecting installation_id from a GitHub app into the main logic which breaks with these changes. 
I'll prepare an updated PR with the needed changes

**Review by okotek — APPROVED** [review]

**Pull request merged**